milabench/validation/planning.py

```diff
@@ -0,0 +1,92 @@
+from collections import defaultdict
+from dataclasses import dataclass, field
+
+from .validation import ValidationLayer
+import voir.instruments.gpu
+
+
+@dataclass
+class Planning:
+    errors: int = 0
+    method: str = None
+    njobs: int = None
+    loss: dict = field(default_factory=dict)
+
+
+class _Layer(ValidationLayer):
+    """Makes sure the events we are receiving are consistent with the planning method
+
+    Notes
+    -----
+    Check that we are receiving loss from the right number of processes
+
+    """
+
+    def __init__(self, **kwargs) -> None:
+        gpus = voir.instruments.gpu.get_gpu_info()["gpus"]
+        self.gpus = len(gpus)
+        self.configs = defaultdict(Planning)
+
+    def on_event(self, entry):
+        if entry.pipe != "data":
+            return
+
+        tag = entry.tag
+        benchname = entry.tag.split(".")[0]
+
+        cfg = entry.pack.config
+        plan = cfg["plan"]
+        method = plan["method"].replace("-", "_")
+        njobs = plan.get("n", 0)
+
+        p = self.configs[benchname]
+
+        if p.njobs is None:
+            p.njobs = njobs
+
+        if p.method is None:
+            p.method = method
+
+        p.errors += int(p.method != method)
+        p.errors += int(p.njobs != njobs)
+        p.errors += int(method not in ("per_gpu", "njobs"))
+
+        # Counts the number of loss we are receiving
+        if entry.data:
+            loss = entry.data.get("loss")
+            if loss is not None:
+                p.loss[tag] = True
```

Comment:
Okay, I see that the presence of a loss is tracked here. It wouldn't be too much to also report it in the loss validator, though.

Reply:
done